### ISSUE TITLE
14$: warn on unused label in lambda body

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#944]
```javascript
a.map(q => {q: q.querySelector('[aria-label="Question"]').value})
```
The above code is syntactically valid, but not intended.

The user intended to put parentheses around the arrow function's body:
```javascript
a.map(q => ({q: q.querySelector('[aria-label="Question"]').value}))
````
